File: PlusConsole/Ascii.vb
'Author: SpartanX10000

Imports System.Text.RegularExpressions

Public Class Ascii

    Private _font As Font

    Public Sub New(Optional type As Figlet = Figlet.Standard)
        Me._font = New Font(type)
    End Sub

    Public Function StringToAscii(ByVal value As String) As String
        Dim ascii As String = String.Empty
        For i As Integer = 1 To _font.Height
            For Each car In value
                ascii += Me.GetCharacter(car, i)
            Next
            ascii += Environment.NewLine
        Next
        Return ascii
    End Function

    Public Function GetCharacter(ByVal value As Char, ByVal line As Integer) As String
        Dim i As Integer = Me._font.CommentLines + ((Convert.ToInt32(value) - 32) * Me._font.Height)
        Dim temp As String = Me._font.FileLines(i + line)
        Dim lineend = temp(temp.Length - 1)
        Dim regex As New Regex("\" & lineend & "{1,2}$")
        temp = regex.Replace(temp, "")
        Return temp.Replace(Me._font.HardBlank, " ")
    End Function

End Class

File: PlusConsole/Ascii.cs
//--------------------------------------------------------------------------------------------------------------------
// Author: Carlos Daniel Angulo López (SpartanX10000)
// Project: PlusConsole
// Class: Ascii.cs
// Description: This class contains the functions to work with ASCII.
//--------------------------------------------------------------------------------------------------------------------

using System;
using System.Text.RegularExpressions;

namespace PlusConsole
{
    public class Ascii
    {
        private Font _font;

        public Ascii(Figlet type = Figlet.Standard)
        {
            this._font = new Font(type);
        }

        public string StringToAscii(string value)
        {
            string ascii = string.Empty;
            for (int i = 1; i <= _font.Height; i++)
            {
                foreach (var car in value)
                    ascii += this.GetCharacter(car, i);
                ascii += Environment.NewLine;
            }
            return ascii;
        }

        public string GetCharacter(char value, int line)
        {
            int i = this._font.CommentLines + ((Convert.ToInt32(value) - 32) * this._font.Height);
            string temp = this._font.FileLines[i + line];
            var lineend = temp[temp.Length - 1];
            Regex regex = new Regex(@"\" + lineend + "{1,2}$");
            temp = regex.Replace(temp, "");
            return temp.Replace(this._font.HardBlank, " ");
        }
    }
}
